client/src/components/Notification/NotificationsMenuItem.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { Box, Divider, ListItemText, Menu, MenuItem as DropdownMenuItem, Badge } from '@mui/material';
+import { NavLink } from 'react-router-dom';
+import { Notification } from '../../interface/Notification';
+import Avatar from '@mui/material/Avatar';
+import { useStyles } from './useStyles';
+import {
+  getWindowDimensions,
+  getResource,
+  unreadNotifications,
+  formatMMDDYYYY,
+  getDescriptiveType,
+} from '../../helpers/NoficationsMenuItemHelper';
+
+const NotificationsMenuItem = (notifications: [Notification], readNotifications: () => void) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const classes = useStyles();
+  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+    readNotifications();
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = (notification?: Notification) => {
+    setAnchorEl(null);
+  };
+  const renderNotifications = () => {
+    return notifications.length >= 1 ? (
+      [
+        notifications.map((notification, index) => {
+          return (
+            <NavLink key={`notification-${index++}`} className={classes.navbarItem} to={getResource(notification)}>
```

Comment:
This (notification component) could've been componentized to increase code maintainability

Reply:
Componentized. Now in a notificationcomponent.